presto-tests/src/test/java/com/facebook/presto/resourcemanager/TestDistributedQueryResource.java

```diff
@@ -156,4 +158,17 @@ private void assertStateCounts(List<BasicQueryInfo> infos, int expectedFinished,
         assertEquals(running, expectedRunning);
         assertEquals(queued, expectedQueued);
     }
+
+    @Test(invocationCount = 10)
+    public void testGetAllQueryInfoForLimits() throws InterruptedException
+    {
+        runToFirstResult(client, coordinator1, "SELECT * from tpch.sf100.orders");
+        runToQueued(client, coordinator1, "SELECT * from tpch.sf100.orders");
+        runToQueued(client, coordinator1, "SELECT * from tpch.sf101.orders");
+        runToQueued(client, coordinator1, "SELECT * from tpch.sf102.orders");
+
```

Comment:
to make this test reliably, add a condition to check how many queries resource manager is seeing:
i.e.
```
List<Query> queries;
do {
        MILLISECONDS.sleep(100);
        queries = resourceManager.getClusterStateProvider().getAllQueries()  
} while (queries.size() != 4)
```

This will make sure all queries are been reported to resource manager before we execute the /v1/query endpoint and in that case you will get consistent result.